components/resource_control/src/service.rs

```diff
@@ -139,6 +151,180 @@ impl ResourceManagerService {
             }
         }
     }
+
+    async fn load_controller_config(&self) -> Result<RequestUnitConfig, ()> {
+        loop {
+            match self
+                .pd_client
+                .load_global_config(RESOURCE_CONTROL_CONTROLLER_CONFIG_PATH.to_string())
+                .await
+            {
+                Ok((items, _)) => {
+                    match serde_json::from_slice::<ControllerConfig>(items[0].get_payload()) {
+                        Ok(c) => return Ok(c.request_unit),
+                        Err(e) => {
+                            error!("parse controller config failed"; "err" => ?e);
+                            let _ = GLOBAL_TIMER_HANDLE
+                                .delay(std::time::Instant::now() + RETRY_INTERVAL)
+                                .compat()
+                                .await;
+                            continue;
+                        }
+                    }
+                }
+                Err(err) => {
+                    error!("failed to controller config"; "err" => ?err);
+                    let _ = GLOBAL_TIMER_HANDLE
+                        .delay(std::time::Instant::now() + RETRY_INTERVAL)
+                        .compat()
+                        .await;
+                    continue;
+                }
+            }
+        }
+    }
+
+    // upload ru metrics periodically.
+    pub async fn upload_ru_metrics(&self) {
+        let mut last_group_statistics_map: HashMap<String, UploadStatistic> = HashMap::new();
+        // load controller config firstly.
+        let config = self.load_controller_config().await.unwrap_or_default();
+
+        loop {
+            let background_groups: Vec<_> = self
+                .manager
+                .resource_groups
+                .iter()
+                .filter_map(|kv| {
+                    let g = kv.value();
+                    g.limiter.clone().map(|limiter| {
+                        let io_consumed = limiter.get_limit_statistics(ResourceType::Io);
+                        let read_bytes_consumed = io_consumed.read_consumed;
+                        let write_bytes_consumed = io_consumed.write_consumed;
+                        let cpu_consumed = limiter
+                            .get_limit_statistics(ResourceType::Cpu)
+                            .total_consumed;
+
+                        let read_total = (config.read_cpu_ms_cost * cpu_consumed as f64
+                            + config.read_cost_per_byte * read_bytes_consumed as f64)
+                            as u64;
+                        let write_total =
+                            (config.write_cost_per_byte * write_bytes_consumed as f64) as u64;
+
+                        (
+                            g.group.name.clone(),
+                            UploadStatistic {
```

Comment:
should keep the version field to handle the case that the statistics data is reset when the quota limiter is recreated.

Reply:
Done. check io and cpu version.